fe/fe-core/src/main/java/com/starrocks/statistic/AnalyzeManager.java

```diff
@@ -263,6 +271,102 @@ public void clearStatisticFromDroppedTable() {
         dropHistogramStatsMetaAndData(statsConnectCtx, tableIdHasDeleted);
     }
 
+    public void dropPartition(long partitionId) {
+        dropPartitionIds.add(partitionId);
+    }
+
+    public void clearStatisticFromDroppedPartition() {
+        checkAndDropPartitionStatistics();
+        dropPartitionStatistics();
+    }
+
+    private void dropPartitionStatistics() {
+        if (dropPartitionIds.isEmpty()) {
+            return;
+        }
+
+        ConnectContext statsConnectCtx = StatisticUtils.buildConnectContext();
+        statsConnectCtx.setThreadLocalInfo();
+
+        List<Long> pids = dropPartitionIds.stream().limit(Config.expr_children_limit / 2).collect(Collectors.toList());
+
+        StatisticExecutor executor = new StatisticExecutor();
+        statsConnectCtx.setThreadLocalInfo();
+        if (executor.dropPartitionStatistics(statsConnectCtx, pids)) {
+            pids.forEach(dropPartitionIds::remove);
```

Comment:
emm, IDE hint `removeAll` may be bad performance,  `A.removeAll(B)`the implement like:
```
for (auto a : A) {
    if (B.contains(a)) {
         A.remove(a)
    }
}
```

I think because the size of A may is greate than B is a high probability